### PR TITLE
Run-time system test for keepout_filter

### PR DESCRIPTION
**Pull request by AlexeyMerzlyakov**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #1263, #1882 |
| Primary OS tested on | Ubuntu 20.04 with Foxy onboard |
| Robotic platform tested on | Gazebo simulation of TB3 |

---

## Description of contribution in a few bullet points

* Added run-time full stack navigation2 test for verification of keepout zones filter enabled in local and global costmaps.
* The test is simply adding one keepout zone placed on the path from point A to point B
* The test is continuously checking current robot position (`/amcl_pose`) and global path plan (`/plan`) each time when they have been updated. Both of them should never intersect / fall into keepout zone.
* The test is made as a basis of `nav2_system_tests/src/system` test. All files were initially copied from this directory into `nav2_system_tests/src/costmap_filters`. Then for readability, all Keepout-related changes in the python-code were made in a separate bcc7fd6310b04edfb91e01e839c35a578cca2339 commit.
* Removed old system test as unnecessary.

### CODE REVIEW

**Comment by AlexeyMerzlyakov** [comment]
This is the first attempt to make run-time python-based full stack verification system test for keepout filter. Please take a look on this. I've tried to make test simply as possible over system one. So criticism is welcoming :smiley:

File: nav2_system_tests/params/keepout_params.yaml [hunk 1]
```diff
@@ -1,7 +1,306 @@
...
+    beam_skip_distance: 0.5
+    beam_skip_error_threshold: 0.9
+    beam_skip_threshold: 0.3
+    do_beamskip: false
+    global_frame_id: "map"
+    lambda_short: 0.1
+    laser_likelihood_max_dist: 2.0
+    laser_max_range: 100.0
+    laser_min_range: -1.0
+    laser_model_type: "likelihood_field"
```
Comment (SteveMacenski) [inline nav2_system_tests/params/keepout_params.yaml:19]: Change this to another laser model, that will push up AMCL's smoke coverage.

File: nav2_system_tests/params/keepout_params.yaml [hunk 2]
```diff
@@ -1,7 +1,306 @@
...
-    filter_info_topic: "costmap_filter_info"
+    use_sim_time: true
     type: 0
-    mask_topic: "filter_mask"
+    filter_info_topic: "/costmap_filter_info"
+    mask_topic: "/filter_mask"
     base: 0.0
     multiplier: 1.0
+
+filter_mask_server:
```
Comment (SteveMacenski) [inline nav2_system_tests/params/keepout_params.yaml:301]: Can you put this file in the costmap filter's directory?

File: nav2_system_tests/src/costmap_filters/test_keepout_launch.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+                              'autostart': 'True'}.items()),
+    ])
+
+
+def main(argv=sys.argv[1:]):
+    ld = generate_launch_description()
+
+    test1_action = ExecuteProcess(
+        cmd=[os.path.join(os.getenv('TEST_DIR'), 'tester_node.py'),
+             '-r', '-2.0', '-0.5', '0.0', '2.0'],
```
Comment (SteveMacenski) [inline nav2_system_tests/src/costmap_filters/test_keepout_launch.py:121]: Do these points actually go through/around that keepout zone?

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

**Comment by SteveMacenski** [comment]
Pull in master, and update was made required to make CI pass

File: nav2_system_tests/params/keepout_params.yaml (hunk 1 above)
Comment (AlexeyMerzlyakov) [reply nav2_system_tests/params/keepout_params.yaml:19]: Do you mean `likelihood_field_prob` ?

Comment (SteveMacenski) [reply nav2_system_tests/params/keepout_params.yaml:19]: yeah, `likelihood_field_prob` would be fine. Or `beam`, I don't really care which. Its just about smoke testing another model if we're specifying a custom param yaml file.

File: nav2_system_tests/src/costmap_filters/test_keepout_launch.py (hunk 3 above)
Comment (AlexeyMerzlyakov) [reply nav2_system_tests/src/costmap_filters/test_keepout_launch.py:121]: Yes, the starting point "A" and finish "B" was remained as in `nav2_system_tests/src/system` tests, and the keepout zone was selected along the path as follows from the picture below:
plan_description

Comment (AlexeyMerzlyakov) [reply nav2_system_tests/src/costmap_filters/test_keepout_launch.py:121]: Maybe, I'd better to attach this picture to testcase as well?

Comment (SteveMacenski) [reply nav2_system_tests/src/costmap_filters/test_keepout_launch.py:121]: Hi, an image would be helpful, yes. I think you should update B to be straight up the aisle way from A. That I'm afraid of is that some small change in initial conditions will make this test not really do its job. Since both paths are close-ish to the same length, its not immediately clear to me that 100% of the time that the "actual path" wouldn't also be the path regardless of the keepout zone. 

If we make the A-B in a straight line up the aisle, then we can add a keepout block in the aisle between 2 of the pillars to force the robot into the left aisle or around the edge. That's a clear case of keepout working because it would never do that with a value straight line path.

tl;dr, I think this would be more robust to changes into the future if we:
- Moved B to straight up from A
- Moved the keepout between 2 pillars in the straightline path

Comment (AlexeyMerzlyakov) [reply nav2_system_tests/src/costmap_filters/test_keepout_launch.py]: Okay, done in https://github.com/ros-planning/navigation2/pull/2030/commits/8d12009621566b955c0bd1ec0be14fe93154c3c8

**Review by SteveMacenski — APPROVED** [review]
